app/src/main/java/com/sunasterisk/movie19/ui/playvideo/PlayVideoViewModel.kt

```diff
@@ -0,0 +1,40 @@
+package com.sunasterisk.movie19.ui.playvideo
+
+import com.google.android.youtube.player.YouTubeInitializationResult
+import com.google.android.youtube.player.YouTubePlayerView
+import com.sunasterisk.movie19.base.BaseViewModel
+import com.sunasterisk.movie19.data.repository.MovieTrailerRepository
+import io.reactivex.rxjava3.android.schedulers.AndroidSchedulers
+import io.reactivex.rxjava3.disposables.CompositeDisposable
+import io.reactivex.rxjava3.kotlin.addTo
+import io.reactivex.rxjava3.schedulers.Schedulers
+
+class PlayVideoViewModel(private val repository: MovieTrailerRepository) : BaseViewModel() {
+
+    override val compositeDisposable: CompositeDisposable = CompositeDisposable()
+
+    fun getVideo(movieId: Int, playerView: YouTubePlayerView) {
+        repository.getVideoTrailer(movieId)
+            .subscribeOn(Schedulers.io())
+            .observeOn(AndroidSchedulers.mainThread())
+            .subscribe {
+                playerView.initialize("AIzaSyACM0OgD8a_bmFOzKP1HiHLOg_Crp-QCsI",
+                    object : com.google.android.youtube.player.YouTubePlayer.OnInitializedListener {
+                        override fun onInitializationSuccess(
+                            p0: com.google.android.youtube.player.YouTubePlayer.Provider?,
+                            p1: com.google.android.youtube.player.YouTubePlayer?,
+                            p2: Boolean
+                        ) {
+                            p1?.loadVideo(it.results[0].key)
+                        }
+
+                        override fun onInitializationFailure(
+                            p0: com.google.android.youtube.player.YouTubePlayer.Provider?,
+                            p1: YouTubeInitializationResult?
```

Comment:
```suggestion
                            provider: YouTubePlayer.Provider?,
                            error: YouTubeInitializationResult?
```
Xử lý lỗi ở đây như nào vậy em?